core/sdk/src/extn/extn_id.rs

```diff
@@ -295,6 +295,20 @@ impl ExtnId {
         false
     }
 
+    /// Checks if the given capability is a badger channel.
+    /// # Examples
+    /// ```
+    /// use ripple_sdk::extn::extn_id::{ExtnId,ExtnClassId};
+    ///
+    /// let badger_channel = ExtnId::new_channel(ExtnClassId::Gateway, "badger".into());
+    /// assert!(badger_channel.is_badger_channel());
+    /// ```
+    pub fn is_badger_channel(&self) -> bool {
```

Comment:
Not sure where this is needed but can we do checks without using bespoke code like this? E.g. by parsing the ID etc.?